client/src/store/utils/reducerFunctions.js

```diff
@@ -109,4 +109,16 @@ export const incrementUnreadMessages = (state, conversationId) => {
       return convo;
     }
   });
+};
+
+// re order conversations when there is a new message, if conversation is not at index 0
+export const updateConvosOrder = (state, conversationId) => {
+  const updatedConvoIndex = state.findIndex(convo => convo.id === conversationId);
+  const newState = [...state];
+  const updatedConvo = newState.filter(convo => convo.id === conversationId);
+  newState.splice(updatedConvoIndex, 1);
+  newState.unshift(updatedConvo[0])
```

Comment:
`unshift` is an O(N) operation in time complexity which is a bit expensive, especially considering that the app could escalate.